apps/backend/models/forumPost.ts

```diff
@@ -0,0 +1,86 @@
+import { list } from '@keystone-6/core';
+import { text, timestamp, relationship } from '@keystone-6/core/fields';
+
+export const ForumPost = list({
+  access: {
+    operation: {
+      query: ({ session }) => !!session,
+      create: ({ session }) => !!session,
+      update: ({ session }) => !!session,
+      delete: ({ session }) => !!session,
+    },
+    filter: {
+      // only logged-in user can see posts
+      query: ({ session }) => {
+        if (!session?.data?.id) return false;
+        return true;
+      },
+    },
+    item: {
+      //only logged-in users can update and delete their own posts
+      update: ({ session, item }) => {
+        if (!session?.data?.id) return false;
+        return session?.data?.id === item.userId;
```

Comment:
```suggestion
        return session?.data?.id === item.authorId;
```

Looks like there is an access control related bug here – your code is referencing `item.userId` but the actual field name is `authorId` (as seen in the Prisma schema)

Suggested fix above